panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java

```diff
@@ -9183,6 +9184,68 @@ public ModelAndView getView(Object o, BindException errors)
         }
     }
 
+    @RequiresPermission(ReadPermission.class)
+    @RequiresLogin
+    public class MyDataViewAction extends SimpleViewAction<MyDataForm>
+    {
+        @Override
+        public void addNavTrail(NavTree root)
+        {
+            root.addChild("My Panorama Public Data");
+        }
+
+        @Override
+        public ModelAndView getView(MyDataForm form, BindException errors) throws Exception
+        {
+            if (form.getUserId() == null)
+            {
+                errors.addError(new LabKeyError("Did not find a user id in the request"));
+                return new SimpleErrorView(errors);
+            }
+            User user = UserManager.getUser(form.getUserId());
+            if (user == null)
+            {
+                errors.addError(new LabKeyError("Could not find a user for id: " + form.getUserId()));
+            }
+            else if (!user.equals(getUser()))
+            {
+                errors.addError(new LabKeyError("User did not match. You cannot view datasets for other users"));
+            }
+            if (errors.hasErrors())
+            {
+                return new SimpleErrorView(errors);
+            }
+
+            QuerySettings settings = new QuerySettings(getViewContext(),  MyDataTableInfo.NAME, MyDataTableInfo.NAME);
+            settings.setContainerFilterName(ContainerFilter.Type.CurrentAndSubfolders.name());
+            QueryView view = new QueryView(new PanoramaPublicSchema(getUser(), getContainer()), settings, errors);
+            view.setTitle("Panorama Public Experiments");
+            view.setShowRecordSelectors(false);
+            view.setShowExportButtons(false);
+            view.setShowDetailsColumn(false);
+            view.setShowDeleteButton(false);
+            view.setShowUpdateColumn(false);
+            view.setShowInsertNewButton(false);
+            view.disableContainerFilterSelection();
+            return view;
+        }
+    }
+
+    public static class MyDataForm
```

Comment:
Do we need this form at all? It seems like we only support viewing data for the current user.

Reply:
No, we don't need a userId since we are only displaying the data for the currently logged in user!  I removed the form.